Marvelo/Libs/Characters/Sources/CharactersViewController.swift

```diff
@@ -36,6 +36,12 @@ public class CharactersViewController: UIViewController {
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
+
+    deinit {
+        Task { @MainActor in
+            viewModel.stopMonitoringDataStore()
```

Comment:
Tasks handling async streams must be manually cleaned up to avoid memory leaks

See https://forums.swift.org/t/task-async-retain-cycles/49585/2 for some discussion and https://www.swiftbysundell.com/articles/memory-management-when-using-async-await/